### PR TITLE
[prometheus-kafka-exporter] Use helm test hooks

### LINKED ISSUE #475: [prometheus-kafka-exporter] test folder pods gets created
**Describe the bug**
It is impossible to disable `kafka` and `zookeeper` test pods, since there are no conditionals or helm hooks configured in https://github.com/prometheus-community/helm-charts/tree/main/charts/prometheus-kafka-exporter/templates/tests

**Version of Helm and Kubernetes**:

Helm Version:

```console
$ helm version
version.BuildInfo{Version:"v3.4.1", GitCommit:"c4e74854886b2efe3321e185578e6db9be0a6e29", GitTreeState:"clean", GoVersion:"go1.14.11"}
```

Kubernetes Version:

```console
$ kubectl version
Server Version: version.Info{Major:"1", Minor:"18+", GitVersion:"v1.18.9-eks-d1db3c", GitCommit:"d1db3c46e55f95d6a7d3e5578689371318f95ff9", GitTreeState:"clean", BuildDate:"2020-10-20T22:18:07Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"linux/amd64"}
```

**Which chart**: prometheus-kafka-exporter

**Which version of the chart**: all

**What happened**: `kubectl get pods` shows not only `prometheus-kafka-exporter`, but also test pods `kafka` and `zookeeper`.

**What you expected to happen**: only `prometheus-kafka-exporter` pod should be provisioned.

**How to reproduce it** (as minimally and precisely as possible):
I use helmfile, but i guess it could be reproduced with bare helm: `helmfile -e <env> apply`

**Changed values of values.yaml** (only put values which differ from the defaults):

```console
fullnameOverride: "prometheus-kafka-exporter"

kafkaServer:
 - one
 - two

service:
  annotations:
    prometheus.io/scrape: "true"
    prometheus.io/port: "9308"

liveness:
  enabled: true

readiness:
  enabled: true
```

### PR DESCRIPTION
**Pull request by Minutis**
#### What this PR does / why we need it:
Do not deploy test `kafka` and `zookeeper` pods in production.

#### Which issue this PR fixes
  - fixes #475 

#### Checklist
- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Review by gkarthiks — DISMISSED** [review]

**Comment by Minutis** [comment]
I am not sure how to fix pipeline. Any help is appreciated. I do not have any experience with chart testing.

**Comment by monotek** [comment]
I'm not sure whats the prefered way here.

In another project i used a helm dependency for ci. See: https://github.com/kokuwaio/helm-charts/tree/main/charts/mysqldump
But might be overkill to add a chart dependency, even if disabled by default, just for ci.

On the other hand kafka used through a helm chart should work with complete configuration out of the box. Should even solve the zookeeper config, as seen in: https://artifacthub.io/packages/helm/bitnami/kafka

File: charts/prometheus-kafka-exporter/Chart.yaml [hunk 1]
```diff
@@ -1,9 +1,9 @@
-apiVersion: v1
+apiVersion: v2
 appVersion: "v1.2.0"
 description: A Helm chart to export the metrics from Kafka in Prometheus format using the kafka-exporter from https://github.com/danielqsj/kafka_exporter
 name: prometheus-kafka-exporter
 home: https://github.com/danielqsj/kafka_exporter
-version: 0.2.0
+version: 0.2.1
```
Comment (monotek) [inline charts/prometheus-kafka-exporter/Chart.yaml:6]: As helm v2 api is used this is a breaking change and new chart version should be 1.0

File: charts/prometheus-kafka-exporter/values.yaml [hunk 2]
```diff
@@ -16,6 +16,10 @@ image:
   tag: latest
   pullPolicy: IfNotPresent
 
+# If enabled Kafka dependency chart will be used. Should be disabled in production environments.
+kafka:
+  enabled: true
```
Comment (monotek) [inline charts/prometheus-kafka-exporter/values.yaml:21]: This should be false by default.
It should be enabled only through ci like in my example.

File: ct.yaml [hunk 3]
```diff
@@ -7,6 +7,7 @@ chart-repos:
   - stable=https://charts.helm.sh/stable
   - prometheus-community=https://prometheus-community.github.io/helm-charts
   - grafana=https://grafana.github.io/helm-charts
+  - kafka=https://charts.bitnami.com/bitnami
```
Comment (monotek) [inline ct.yaml:10]: Please rename repo to bitnami as it also contains other charts.

**Review by monotek — CHANGES_REQUESTED** [review]

**Comment by Minutis** [comment]
Can't get test tun run... Didn't even do anything related to serviceaccounts, still getting:
```
Error: serviceaccounts "prometheus-kafka-exporter-es5m8l6yyp" already exists
```

**Comment by arvidviderberg** [comment]
Hello, we're having the same issue and would really get this PR working. Can we help in any way? 🙂

**Comment by Minutis** [comment]
Problem is with testing - I can not get these tests to pass.

**Comment by arvidviderberg** [comment]
Maybe @gkarthiks or @golgoth31 can assist? 🙂

**Comment by golgoth31** [comment]
Hi all,
for the last run, we got this error:
Error: services "prometheus-kafka-exporter-gdy24x19ba" already exists

**Comment by golgoth31** [comment]
perhaps the check action is a bit old:
uses: helm/chart-testing-action@v1.0.0
while the current version is v2.0.1: https://github.com/helm/chart-testing-action/releases

**Comment by monotek** [comment]
With the changes i've pushed it works locally.

Neverthelless the current error could be indeed be a problem of to old chart testing action, as lookup function was added to helm 3.1(?) and i'm not sure which helm version is used in the action.

@scottrigby @torstenwalter 
I guess we should update the ci pipline.
I've created a pr for this: https://github.com/prometheus-community/helm-charts/pull/611

**Review by monotek — APPROVED** [review]

**Comment by gkarthiks** [comment]
I am just waiting for admins to confirm the changes are good to go across all charts.

**Comment by monotek** [comment]
If #611 is merged first and this one is rebased were good to go too :)

**Review by monotek — DISMISSED** [review]

**Comment by monotek** [comment]
@prometheus-community/helm-charts-admins 
we have a change in the ct.yaml here so i guess on of you guys has to approve and merge...

**Review by monotek — APPROVED** [review]

**Comment by monotek** [comment]
@gkarthiks 
i've removed the change in the ct.yaml as bitnami repo was already inside so you should be able to merge it now :)

**Review by gkarthiks — APPROVED** [review]
amazing changes for stabilization. Thank you all for the effort towards this. 🚀

**Pull request merged**